tests/drivers/gpio/gpio_basic_api/dts/bindings/test,gpio_basic_api.yaml

```diff
@@ -0,0 +1,21 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+title: Hardware resources required for the gpio_basic_api test.
+
+description: >
+    This binding provides resources required to build and run the
+    tests/drivers/gpio/gpio_basic_api test in Zephyr.
+    
+compatible: "test,gpio_basic_api"
+
+properties:
+    gpios:
```

Comment:
would it make sense for this to be:

```
	resources {
		compatible = "test,gpio_basic_api";
		/* Arduino D0, D1 */
		in-gpios = <&gpiof 6 0>;
        out-gpios = <&gpiof 5 0>;
	};
```

Reply:
Yes.  Updated PR, and added check for incompatible controllers.